phlib/include/phnativeinl.h

```diff
@@ -820,24 +820,51 @@ FORCEINLINE
 NTSTATUS
 PhGetProcessAppMemoryInformation(
     _In_ HANDLE ProcessHandle,
-    _Out_ PPROCESS_JOB_MEMORY_INFO JobMemoryInfo
+    _Out_ PAPP_MEMORY_INFORMATION AppMemoryInfo
     )
 {
     NTSTATUS status;
-    PROCESS_JOB_MEMORY_INFO jobMemoryInfo;
+    APP_MEMORY_INFORMATION appMemoryInfo = {0};
+    VM_COUNTERS_EX2 vmCounters;
 
-    // Win32 called this ProcessAppMemoryInfo with APP_MEMORY_INFORMATION struct (dmex)
     status = NtQueryInformationProcess(
         ProcessHandle,
-        ProcessJobMemoryInformation,
-        &jobMemoryInfo,
-        sizeof(PROCESS_JOB_MEMORY_INFO),
+        ProcessVmCounters,
+        &vmCounters,
+        sizeof(vmCounters),
         NULL
-        );
+    );
+
+    if (NT_SUCCESS(status))
+    {
+        appMemoryInfo.PrivateCommitUsage = vmCounters.CountersEx.PagefileUsage;
+        appMemoryInfo.TotalCommitUsage = appMemoryInfo.PrivateCommitUsage + vmCounters.SharedCommitUsage;
+        appMemoryInfo.PeakPrivateCommitUsage = vmCounters.CountersEx.PeakPagefileUsage;
+    }
+    else
+    {
+        // fall-back to Job memory info on older systems
```

Comment:
Why use the query failure as an indication that the info query is not supported on a platform? If we know where is began support we should check `WindowsVersion` and make the appropriate call.

It is also unclear to me if we should even keep querying `ProcessJobMemoryInformation` if it's inaccurate.

Reply:
> If we know where is began support we should check WindowsVersion and make the appropriate call.

I'm not sure when support for `VM_COUNTERS_EX2` was added exactly.
Do you know?

> Should we even keep querying ProcessJobMemoryInformation if it's inaccurate?

Yup, that's right. I'd personally remove it.
I just assumed you guys would like to keep at least some info on older systems (if it was like that for long time).

I'd gladly remove the fallback. Just let me know if it should stay or not.

Reply:
> I'm not sure when support for VM_COUNTERS_EX2 was added exactly.
Do you know?

Just checked, it was added between Windows 8.1 and Windows 10 TH1.

Reply:
> Just checked, it was added between Windows 8.1 and Windows 10 TH1.

Thanks.
I guess there should be no need for exact OS version check.
We can check for `STATUS_INFO_LENGTH_MISMATCH` that the API returns when new (unsupported) structure is supplied.

@jxy-s 
Do we want to keep the (improper) fallback? Or rather just remove it?

Reply:
I'm okay with removing it, it seems errant to me. But I'd like input from @dmex since he's the author.